### ISSUE TITLE
a11y: Checkbox group name not announced

### ISSUE THREAD

**Issue by anezthes** [vaadin/vaadin-checkbox#168]
Typically, checkbox groups are structured like this:
```
<fieldset>
  <legend>Consoles</legend>
  <div>
    <input type="checkbox" name="consoles" id="ps">
    <label for="ps">PlayStation</label>
  </div>
  <div>
    <input type="checkbox" name="consoles" id="xbox">
    <label for="xbox">Xbox</label>
  </div>
  <div>
    <input type="checkbox" name="consoles" id="wii">
    <label for="wii">Wii</label>
  </div>
</fieldset>
```

Tabbing into and focusing the first checkbox, VoiceOver states:
```
PlayStation, unticked, Tick box, Consoles, group
```

With the current structure of `vaadin-checkbox-group` the name of the group is never announced (tested with Chrome/Safari).

**Solution**
Use fieldset and legend?

**Comment by anezthes** [comment]
We could work around the for/id issue with:

```
<fieldset>
  <legend>Consoles</legend>
  <label>
    <input type="checkbox" name="consoles">
    PlayStation
  </label>
  <label>
    <input type="checkbox" name="consoles">
    Xbox
  </label>
  <label>
    <input type="checkbox" name="consoles">
    Wii
  </label>
</fieldset>
```

**Comment by anezthes** [comment]
Tested the following structure with great success in Chrome and Safari using VoiceOver:

**`vaadin-checkbox-group`**
```
<fieldset>
  <legend>${this.legend}</legend>
  <slot></slot>
</fieldset>
```
**`vaadin-checkbox`**
```
<label>
  <input type="checkbox">
  <span>
    <slot></slot>
  </span>
</label>
```

One thing to keep in mind is that VoiceOver will pronounce the legend twice when a checkbox in the group receives keyboard focus in Safari. The legend is also pronounced (once) when the focus within the group changes.

We could consider using `aria-labelledby` and `role="group"` but it feels like `fieldset` and `legend` is the safer option moving forward.

**Comment by juuso-vaadin** [comment]
Required groups are not announced as name of the group isn't announced.

**Comment by web-padawan** [comment]
Closing in favor of https://github.com/vaadin/web-components/issues/94 which describes the problem in more detail.
